feat(react-grid): add scroll to row in virtual table (#2511)
DevExpress/devextreme-reactive: packages/dx-react-grid/docs/guides/virtual-scrolling.md

If the Grid should have the same size as the container element, set the `VirtualTable` plugin's `height` property to "auto" and the Grid root element's style setting to `height: 100%`.

.embedded-demo({ "path": "grid-virtual-scrolling/stretching-to-parent-element", "showThemeSelector": true })

## Scroll to Row

> The virtual table can be scrolled programmatically, through the `scrollToIndex` and `scrollToId` methods. To call these, you should use `VirtualTable` plugin's reference. The following demo demonstrates how to do that:

Reviewer: We should add a link to VirtualTable's Methods section

.embedded-demo({ "path": "grid-virtual-scrolling/stretching-to-parent-element", "showThemeSelector": true })

## Scroll to Row

> The virtual table can be scrolled programmatically, through the `scrollToRow` methods. To call these, you should use `VirtualTable` plugin's reference. The following demo demonstrates how scroll table to added row :

Reviewer 2: `The following demo demonstrates how scroll table to added row` ->
`The following demo demonstrates how scroll table to new and added rows`

Author: Corrected

> The virtual table can be scrolled programmatically, through the `scrollToRow` methods. To call these, you should use `VirtualTable` plugin's ref property. The following demo demonstrates how scroll table to new and added rows:

Reviewer 3: ```suggestion
To scroll the table to a particular row, call the `scrollToRow` method. It accepts a row ID. To call the method, you need the `VirtualTable` plugin's ref.

In the following demo, the `scrollToRow` method is used to scroll the table to a new or saved row. When you add a new row, it is added to the table's top, and the table is scrolled to it. When you save the row, its position is changed according to sorting, and the table is scrolled to that position.
```

.embedded-demo({ "path": "grid-virtual-scrolling/scroll-to-row", "showThemeSelector": true })

> Note that the virtual table with Lazy Loading doesn't allow to scroll to row's id.

Reviewer 3: ```suggestion
NOTE: Scrolling to a row does not work with lazy loading. This feature requires the IDs of all rows, but in lazy loading mode, the Grid loads rows in parts.
```

Reviewer 3: Please make the phrase "lazy loading" a link to [this article](https://devexpress.github.io/devextreme-reactive/react/grid/docs/guides/lazy-loading/#react-grid---virtual-scrolling-with-remote-data-lazy-loading).

.embedded-demo({ "path": "grid-virtual-scrolling/stretching-to-parent-element", "showThemeSelector": true })

## Scroll to Row

> To scroll the table to a particular row, call the `scrollToRow` method. It accepts a row ID. To call the method, you need the `VirtualTable` plugin's ref.

Reviewer 4: Perhaps:

"To scroll the table to a particular row, call the `scrollToRow` method and pass the row ID as its parameter"?

Reviewer 3: ```suggestion
To scroll the table to a particular row, call the `scrollToRow` method and pass the row ID as its parameter. To call the method, you need the `VirtualTable` plugin's ref.
```

> In the following demo, the `scrollToRow` method is used to scroll the table to a new or saved row. When you add a new row, it is added to the table's top, and the table is scrolled to it. When you save the row, its position is changed according to sorting, and the table is scrolled to that position.

Reviewer 4: ```suggestion
In the following demo, the `scrollToRow` method is used to scroll the table to a new or saved row. When you add a new row, it is added to the top of the table, and the table is scrolled to it. When you save the row, its position is changed according to sorting, and the table is scrolled to that position.
```

.embedded-demo({ "path": "grid-virtual-scrolling/scroll-to-row", "showThemeSelector": true })

> NOTE: Scrolling to a row does not work with [lazy loading](./lazy-loading.md/#react-grid---virtual-scrolling-with-remote-data-lazy-loading). This feature requires the IDs of all rows, but in lazy loading mode, the Grid loads rows in parts.

Reviewer 4: ```suggestion
NOTE: Scrolling to a row cannot be used with [lazy loading](./lazy-loading.md/#react-grid---virtual-scrolling-with-remote-data-lazy-loading). This is because the Grid loads rows in parts in lazy loading mode and this feature requires all the row IDs.
```

Reviewer 3: ```suggestion
NOTE: Scrolling to a row cannot be used with [lazy loading](./lazy-loading.md/#react-grid---virtual-scrolling-with-remote-data-lazy-loading). This is because the Grid loads rows in parts in lazy loading mode, and scrolling to a row requires all the row IDs.
```
